### PR TITLE
OCPBUGS-29531: Add include.release.openshift.io/hypershift label

### PR DESCRIPTION
**Pull request by danielmellado**
* [ ] I added CHANGELOG entry for this change.
* [X] No user facing changes, so no entry in CHANGELOG was needed.

### CODE REVIEW

**Review by marioferh — COMMENTED** [review]
/lgtm

**Comment by danielmellado** [comment]
@bburt-rh mind taking a look at this for the docs-approved label? thanks!

**Comment by ashwindasr** [comment]
/hold

@danielmellado Looks like the annotation is defined in other places in the code base as well. Can you check this [diff](https://github.com/danielmellado/cluster-monitoring-operator/compare/danielmellado:add_hypershift_annotation...openshift-art-build-bot:art-8431-test?expand=1) and pull this in as well if possible?

**Comment by danielmellado** [comment]
Hi @ashwindasr I've identified a few more places where this was used. Regarding the diff I just see the same files I had, not sure if you wanted to show something else! I can't just pull it in as the manifests are generated from the jsonnet files and they would be overridden. Let me know!

**Comment by ashwindasr** [comment]
Thanks @danielmellado !

As long as `include.release.openshift.io/hypershift': 'true'` follows all instances of `'include.release.openshift.io/ibm-cloud-managed': 'true'` we should be good!

File: jsonnet/components/cluster-monitoring-operator.libsonnet [hunk 1]
```diff
@@ -150,6 +150,7 @@ function(params) {
       name: cfg.name,
       annotations: {
         'include.release.openshift.io/ibm-cloud-managed': 'true',
+        'include.release.openshift.io/hypershift': 'true',
```
Comment (machine424) [inline jsonnet/components/cluster-monitoring-operator.libsonnet:153]: I'd be great if we're able to identify those CVO resources in `add-annotations.libsonnet` and inject the needed annotations there as well.

**Comment by machine424** [comment]
I don't know the effect of the annotation, but
/lgtm

**Comment by machine424** [comment]
Note that not all files from `manifests/` were annotated https://github.com/openshift/cluster-monitoring-operator/tree/master/manifests
/hold
please unhold if it's ok.

**Comment by danielmellado** [comment]
Hm, true, i.e. https://github.com/openshift/cluster-monitoring-operator/blob/master/manifests/0000_50_cluster-monitoring-operator_04-config.yaml#L1060 @simonpasquier would it be ok just to update those manually?

File: jsonnet/components/cluster-monitoring-operator.libsonnet (hunk 1 above)
Comment (danielmellado) [reply jsonnet/components/cluster-monitoring-operator.libsonnet:153]: Agree, I'll tackle the libsonnet cleanup on a different PR to unblock this one. Some files are not generated at all by the jsonnet and that's why they're left ignored. It'd be good to consolidate this.

**Comment by danielmellado** [comment]
@ashwindasr the version gate is expected to fail given the current openshift release status. So let's just merge this as soon as the other gates pass again (they were passing before your retest xD) Thanks!

**Comment by danielmellado** [comment]
CI Issues seem totally unrelated ({Operator degraded (MissingDependency): machine-api not ready  Operator degraded (MissingDependency): machine-api not ready}
) so let's check if we're lucky this time

**Comment by danielmellado** [comment]
@simonpasquier I'm a bit lost here, this was previously passing and tbh I feel that the issues are unrelated to CMO :\

**Comment by danielmellado** [comment]
there was an issue with CI, let's double check now

**Comment by danielmellado** [comment]
this passed... not sure what's going on with the CI, it doesn't even get to deploy the cluster.

**Comment by danielmellado** [comment]
Adding @jparrill and @csrwng, mind taking a look at this CI runs? not sure about the hypershift annotation and I'm aware about the MachineConfigPool recent CI outage but besides that IPI install doesn't even get to finish. Thanks!

**Comment by danielmellado** [comment]
note: Agreed with the ART team to skip a specific ROKS annotation for this effort which was making IPI install phase fail

**Review by marioferh — COMMENTED** [review]
/lgtm

**Pull request merged**